From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: simpleauthority <jacob@algorithmjunkie.com>
Date: Tue, 28 May 2019 03:41:28 -0700
Subject: [PATCH] Add BlockSoundGroup interface

This PR adds the getSoundGroup() method in Block which returns a BlockSoundGroup

diff --git a/src/main/java/com/destroystokyo/paper/block/BlockSoundGroup.java b/src/main/java/com/destroystokyo/paper/block/BlockSoundGroup.java
new file mode 100644
index 0000000000000000000000000000000000000000..ec36942128cbacae171584c89480b4aae3ae3e2f
--- /dev/null
+++ b/src/main/java/com/destroystokyo/paper/block/BlockSoundGroup.java
@@ -0,0 +1,64 @@
+package com.destroystokyo.paper.block;
+
+import org.bukkit.Sound;
+import org.bukkit.block.Block;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Represents the sounds that a {@link Block} makes in certain situations
+ * <p>
+ * The sound group includes break, step, place, hit, and fall sounds.
+ * @deprecated use {@link org.bukkit.SoundGroup}
+ */
+@Deprecated(forRemoval = true)
+public interface BlockSoundGroup {
+    /**
+     * Gets the sound that plays when breaking this block
+     *
+     * @return The break sound
+     * @deprecated use {@link org.bukkit.SoundGroup#getBreakSound()}
+     */
+    @NotNull
+    @Deprecated(forRemoval = true)
+    Sound getBreakSound();
+
+    /**
+     * Gets the sound that plays when stepping on this block
+     *
+     * @return The step sound
+     * @deprecated use {@link org.bukkit.SoundGroup#getStepSound()}
+     */
+    @NotNull
+    @Deprecated(forRemoval = true)
+    Sound getStepSound();
+
+    /**
+     * Gets the sound that plays when placing this block
+     *
+     * @return The place sound
+     * @deprecated use {@link org.bukkit.SoundGroup#getPlaceSound()}
+     */
+    @NotNull
+    @Deprecated(forRemoval = true)
+    Sound getPlaceSound();
+
+    /**
+     * Gets the sound that plays when hitting this block
+     *
+     * @return The hit sound
+     * @deprecated use {@link org.bukkit.SoundGroup#getHitSound()}
+     */
+    @NotNull
+    @Deprecated(forRemoval = true)
+    Sound getHitSound();
+
+    /**
+     * Gets the sound that plays when this block falls
+     *
+     * @return The fall sound
+     * @deprecated use {@link org.bukkit.SoundGroup#getFallSound()}
+     */
+    @NotNull
+    @Deprecated(forRemoval = true)
+    Sound getFallSound();
+}
diff --git a/src/main/java/org/bukkit/block/Block.java b/src/main/java/org/bukkit/block/Block.java
index 653a83b7ceb5cec1d92aed5250ae50e714fe5f9f..2b267d4c9adcc13722d4a655a98ca16a0f9746ca 100644
--- a/src/main/java/org/bukkit/block/Block.java
+++ b/src/main/java/org/bukkit/block/Block.java
@@ -607,4 +607,25 @@ public interface Block extends Metadatable, Translatable {
      * @return <code>true</code> if the block data can be placed here
      */
     boolean canPlace(@NotNull BlockData data);
+
+    // Paper start
+    /**
+     * Gets the {@link com.destroystokyo.paper.block.BlockSoundGroup} for this block.
+     * <p>
+     * This object contains the block, step, place, hit, and fall sounds.
+     *
+     * @return the sound group for this block
+     * @deprecated use {@link #getBlockSoundGroup()}
+     */
+    @NotNull
+    @Deprecated(forRemoval = true)
+    com.destroystokyo.paper.block.BlockSoundGroup getSoundGroup();
+
+    /**
+     * Gets the {@link org.bukkit.SoundGroup} for this block.
+     *
+     * @return the sound group for this block
+     */
+    @NotNull org.bukkit.SoundGroup getBlockSoundGroup();
+    // Paper end
 }
